flight/targets/brainre1/fw/ir_protocols.c

```diff
@@ -0,0 +1,155 @@
+/**
+ ******************************************************************************
+ * @addtogroup dRonin Targets
+ * @{
+ * @addtogroup BrainRE1 support files
+ * @{
+ *
+ * @file       ir_protocols.c
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2016
+ * @brief      Generate packets for various infrared lap timin protocols
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#include <pios.h>
+#include <misc_math.h>
+
+
+/**
+ * @brief Invert bits, needed for some protocols
+ */
+void ir_invert_bits(uint8_t * data, uint8_t data_len)
+{
+	for (int i=0; i<data_len; i++) {
+		data[i] = ~data[i];
+	}
+}
+
+
+/* Based on code generated by pycrc v0.9, https://pycrc.org
+* Note: this is regular crc-16-ccitt CRC
+* using the configuration:
+*    Width         = 16
+*    Poly          = 0x1021
+*    Xor_In        = 0x0000
+*    ReflectIn     = False
+*    Xor_Out       = 0x0000
+*    ReflectOut    = False
+*    Algorithm     = bit-by-bit-fast
+*****************************************************************************/
+uint16_t crc_ilap_trackmate(uint16_t crc, const uint8_t *data, uint8_t data_len)
```

Comment:
This should go into a lib and be called crc16_ccitt or something, this poly is used by other things (offhand Jeti Ex.Bus).